CRM/Contact/BAO/Contact/Permission.php

```diff
@@ -43,8 +133,16 @@ class CRM_Contact_BAO_Contact_Permission {
    *   true if the user has permission, false otherwise
    */
   public static function allow($id, $type = CRM_Core_Permission::VIEW) {
-    $tables = array();
-    $whereTables = array();
+    // get logged in user
+    $session   = CRM_Core_Session::singleton();
```

Comment:
ditto about the preferred function - saves 3 lines here as the function name should be clear enough to justify dropping the comment line :-)